src/cluster_info.rs

```diff
@@ -274,6 +288,21 @@ impl ClusterInfo {
             .collect()
     }
 
+    /// all peers with valid tvus and gossip ports
+    pub fn repair_peers(&self) -> Vec<NodeInfo> {
+        let me = self.my_data().id;
+        self.gossip
+            .crds
+            .table
+            .values()
+            .filter_map(|x| x.value.contact_info())
+            .filter(|x| x.id != me)
+            .filter(|x| ContactInfo::is_valid_address(&x.tvu))
```

Comment:
How about?

```rust
self.retransmit_peers().into_iter().filter(|x| ContactInfo::is_valid_address(&x.gossip)).collect()
```

Reply:
Yep! Better :+1: It should be tvu_peers but I get the point :)